psga/common/primer_autodetection.nf

```diff
@@ -14,24 +14,36 @@ process primer_autodetection {
     tuple path("*_primer.txt"), path(fastq), emit: ch_files
     path "*_primer_data.csv", emit: ch_primer_data
     path "*_primer_detection.csv", emit: ch_primer_coverage
+    path "*_trimming_report.txt", emit: ch_trimming_report
 
   shell:
   '''
-  # select the input fastq file based on illumina or ont.
+  # select the input fastq file based on illumina or ont and trim adaptor sequences.
   # For illumina, 1 read is sufficient for primer autodetection
   if [[ "!{params.sequencing_technology}" == "illumina" ]]; then
     file_1="$(ls *_1.fastq.gz)"
+    file_2="$(ls *_2.fastq.gz)"
+    sample_id=$( echo ${file_1} | cut -d '_' -f1)
+    trimmed_file_1="${sample_id}_1_val_1.fq.gz"
+    trim_galore --paired --gzip ${file_1} ${file_2} > ${sample_id}_trimming_report.txt 2>&1
```

Comment:
NOTE: `trim_galore` is configured as in https://github.com/Congenica/ncov2019-artic-nf/blob/master/modules/illumina.nf#L23 .